Transfer Funds Action Wiring

rdig opened the pull request:
## Description

This PR wires up the _"Move Funds"_ flow that creates a new action from moving funds, inside a colony from a domain to the other.

Apart from that it also refactors the `colonyActions` resolver and static maps in order for this to scale with new actions added.

**Changes** 

- [x] Added `MoveFunds` action `ColonyActions` type
- [x] Added `MoveFunds` action and events copy
- [x] Moved all static maps to a central location
- [x] Refactored `colonyActions` resolver
- [x] Refactored `getActionType` type util to use a map
- [x] Added `getEventsForActions` util
- [x] Refactor `ActionsPage`, `ActionsPageFeed` and they're subcomponents to all share the same data format being passed down
- [x] Added `getActionValues`, `getPaymentActionValues` and `getMoveFundsActionValues` util
- [x] Refactored move funds redux actions and sagas into colony action types and sagas
- [x] Wired up `TransferFundsDialog` in the wizard flow
- [x] Refactor `TransferFundsDialog` to properly display values
- [x] Refactor `TransferFundsDialog` to properly validate values
- [x] Add `TransferFundsDialog` permissions missing errors
- [x] Add proper `ActionsPageFeed` loading state

**Reviewers Note**

Sorry for being so big :(

**Demo**

![cs-demo-move-funds-action](https://user-images.githubusercontent.com/1193222/102341161-1a178b80-3fa0-11eb-9a44-50df3f253292.gif)

Resolves DEV-64

alicjakujawa reviewed (changes requested):
So there are small design issues still here - too big spaces between header and "You need the following permission(s)" label, and at the bottom between message about permissions required - see screenshots. Also "x" doesn't look like the one from designs - Of course we can create issue for those design adjustment if you prefer.
Another problem is about modal - on my computer im not able to see full dialog see screenshots, but again this should be address as separate issue.
<img width="560" alt="1" src="https://user-images.githubusercontent.com/13069555/102380828-b35a9800-3fc8-11eb-83f4-0d1419dfd28b.png">
<img width="547" alt="2" src="https://user-images.githubusercontent.com/13069555/102380834-b48bc500-3fc8-11eb-933c-f6e6c6b502c0.png">
<img width="591" alt="3" src="https://user-images.githubusercontent.com/13069555/102380835-b5245b80-3fc8-11eb-83fd-af949b27a62e.png">

The only thing which is important - confirm button is enable if im sending from domain without funds - see screenshot

<img width="734" alt="Screenshot 2020-12-16 at 17 53 08" src="https://user-images.githubusercontent.com/13069555/102380957-e735bd80-3fc8-11eb-983b-f080779baea5.png">

rdig commented:
> So there are small design issues still here - too big spaces between header and "You need the following permission(s)" label, and at the bottom between message about permissions required - see screenshots. Also "x" doesn't look like the one from designs - Of course we can create issue for those design adjustment if you prefer.

I just used the standard core components, didn't really cared too much about styles. But I can look into it if you want me to.

alicjakujawa commented:
> > So there are small design issues still here - too big spaces between header and "You need the following permission(s)" label, and at the bottom between message about permissions required - see screenshots. Also "x" doesn't look like the one from designs - Of course we can create issue for those design adjustment if you prefer.
> 
> I just used the standard core components, didn't really cared too much about styles. But I can look into it if you want me to.

I mean we can just create task for improving styles if we want stick to designs

rdig commented on src/modules/dashboard/components/ActionsPage/InputStorageWidget/InputStorageWidget.tsx:
```diff
@@ -59,9 +59,9 @@ const InputStorageWidget = () => {
-      submit={ActionTypes.MOVE_FUNDS_BETWEEN_POTS} // @TODO: Add in correct ActionTypes
-      error={ActionTypes.MOVE_FUNDS_BETWEEN_POTS_ERROR}
-      success={ActionTypes.MOVE_FUNDS_BETWEEN_POTS}
+      submit={ActionTypes.COLONY_ACTION_GENERIC} // @TODO: Add in correct ActionTypes
```
I know them, I just don't want to wire them into this branch as well.

It'll be something like: `COLONY_ACTION_RECOVERY_UPDATE_STORAGE_SLOT` 

But that's to be done in DEV-75

For now I just needed some redux actions that exist, in order to make `eslint` stop complaining.

rdig commented on src/modules/dashboard/components/TransferFundsDialog/TransferFundsDialogForm.tsx:
```diff
@@ -104,14 +104,19 @@ const TransferFundsDialogForm = ({
+  const toDomainId = values.toDomain
     ? parseInt(values.toDomain, 10)
-    : ROOT_DOMAIN_ID;
+    : undefined;
```
I need this to be explicitly `undefined`, if I use your suggested change it will end up being `false` if `toDomain` doesn't exist.

rdig commented on src/modules/dashboard/components/TransferFundsDialog/TransferFundsDialogForm.tsx:
```diff
@@ -165,69 +168,76 @@ const TransferFundsDialogForm = ({
+        );
+      return getBalanceFromToken(token, toDomainId);
+    }
+    return undefined;
```
I need to return _something_ to make `eslint` happy, It really doesn't matter what. And I prefere returning something like:
`return undefined;` or `return null;` as opposed of just an undeclared return `return;`

rdig commented on src/modules/dashboard/sagas/actions.ts:
```diff
@@ -126,18 +124,128 @@ function* createPaymentAction({
+
+    yield put<AllActions>({
+      type: ActionTypes.COLONY_ACTION_MOVE_FUNDS_SUCCESS,
+      // payload: { colonyAddress, tokenAddress, fromPot, toPot, amount },
```
Good catch, it's a leftover that needs to be removed.

rdig commented:
> Another problem is about modal - on my computer im not able to see full dialog see screenshots, but again this should be address as separate issue.

We really need to find a good solution for this. Maybe in DEV-118 ?

alicjakujawa commented:
> > Another problem is about modal - on my computer im not able to see full dialog see screenshots, but again this should be address as separate issue.
> 
> We really need to find a good solution for this. Maybe in DEV-118 ?

Yes it would fits there great

rdig commented:
> The only thing which is important - confirm button is enable if im sending from domain without funds - see screenshot
> <Image ... />

How did you get into that state? I cannot replicate it:

![Screenshot from 2020-12-17 12-44-09](https://user-images.githubusercontent.com/1193222/102477905-a475f300-4065-11eb-81b8-a678f0010708.png)

rdig commented:
@alicjakujawa 
- updated error messages alignments
- improved the way available funds / select errors overlap with each other
- removed commented out code

Ready for another look-over

alicjakujawa commented:
<img width="781" alt="Screenshot 2020-12-17 at 13 07 25" src="https://user-images.githubusercontent.com/13069555/102486161-e81e2c00-4068-11eb-8853-bf77f811d34e.png">
@rdig So, i Select set "from" to domain without funds, set to - to root domain (at this point confirm was disabled) BUT when i type to input value, confirm button was enable again

ArmandoGraterol reviewed (approved):
It's lit, it's lit :fire::ok_hand:

rdig commented:
@alicjakujawa Still can't replicate your issue. Here's a demo of my following the steps you described:

![demo-move-funds-insufficient-pot](https://user-images.githubusercontent.com/1193222/102625654-77e9d600-414e-11eb-8ee9-bed076d8c5de.gif)

Are you sure you are using the latest commit from the branch?

I can investigate further, but without a proper case to replicate it's going to be very hard.

rdig commented:
@alicjakujawa I've refactored the way that modal is validated. Not it validates:
- on mount
- on every domain change -- both from and to
- on amount change

Please pull the latest code and test it again locally.

Thanks :+1:

alicjakujawa reviewed (approved):
Dont know, what you changed but now is working fine 🔥
